src/main/java/duke/Command.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Represents a given command by a user in Duke. A command is consists of 2 parts,
+ * keyword and content, both of which are String type.
+ */
+public class Command {
+    private final String keyword;
+    private final String content;
+    private final String EXIT = "bye";
+    private final String STATUS_COMPLETE = "done";
+    private final String STATUS_INCOMPLETE = "not done";
+
+    public Command(String keyword, String content) {
+        this.keyword = keyword.toLowerCase();
+        this.content = content;
+    }
+
+    /**
+     * Returns the ID of a Command object in int type.
+     * @return Command ID
+     */
+    protected int getContentId() {
+        int index = content.indexOf(' ');
+        if (index > 0) {
+            return Integer.parseInt(content.substring(0, index));
+        }
+        return Integer.parseInt(content);
+    }
+
+    protected void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (Parser.isTaskKeyword(keyword)) {
+            Parser.isValidTaskCommand(keyword, content);
+            if (keyword.equals("todo")) {
+                Task newTask = new Todo(content, 'T');
+                tasks.add(newTask);
+                ui.addTaskConfirmation(newTask, tasks.getSize());
+            } else if (keyword.equals("deadline")) {
+                Task newTask = new Deadline(content, 'D');
+                tasks.add(newTask);
+                ui.addTaskConfirmation(newTask, tasks.getSize());
+            } else if (keyword.equals("event")) {
+                Task newTask = new Event(content, 'E');
+                tasks.add(newTask);
+                ui.addTaskConfirmation(newTask, tasks.getSize());
+            }
+        } else {
+            if (keyword.equals("list")) {
+                ui.display(tasks.enumerateList());
+            } else if (keyword.equals("find")){
+                ui.display(tasks.find(content));
+            } else if (keyword.equals(EXIT)) {
+                ui.lineBreak();
+                ui.outro();
+            } else {
+                int index = getContentId() - 1;
+                if (keyword.equals("mark")) {
+                    Task updatedTask = tasks.markTask(index);
+                    ui.updateTask(updatedTask, STATUS_COMPLETE);
+                } else if (keyword.equals("unmark")) {
+                    Task updatedTask = tasks.unmarkTask(index);
+                    ui.updateTask(updatedTask, STATUS_INCOMPLETE);
+                } else if (keyword.equals("delete")) {
+                    Task deletedTask = tasks.delete(index);
+                    ui.deleteTaskConfirmation(deletedTask, tasks.getSize());
+                }
+            }
+        }
+        storage.save(tasks.enumerateList());
+    }
+
+    /**
+     * Check whether a given Command instance is consists of the keyword bye (non case-sensitive).
+     * @return true if the command instance is a bye command, false otherwise.
+     */
+    protected boolean isBye(){
```

Comment:
`isBye` seems vague. Perhaps you could rename it to `isByeCommand`.
```suggestion
    protected boolean isByeCommand(){
```